### PR TITLE
Added the ability to download 'Recording' files.

### PR DESCRIPTION
**Pull request by CameronRP**
Also fixed a bug with not accepting users JWTs when getting recordings.

### CODE REVIEW

File: api/V1/Recording.js [hunk 1]
```diff
@@ -161,7 +162,7 @@ module.exports = (app, baseUrl) => {
       log.info(request.method + " Request: " + request.url);
 
       // Check that the request was authenticated by a User.
-      if (requestUtil.isFromAUser(request))
+      if (!requestUtil.isFromAUser(request))
```
Comment (mjs) [inline api/V1/Recording.js]: oops :)

File: api/V1/Recording.js [hunk 2]
```diff
@@ -216,4 +217,80 @@ module.exports = (app, baseUrl) => {
         rows: result.rows,
       });
     });
+
+    /**
+    * @api {get} /api/v1/recordings/:id Get a recording
+    * @apiName GetRecording
+    * @apiGroup Recordings
+    * @apiDescription This call returns a recording in a JSON format and a JSON
```
Comment (mjs) [inline api/V1/Recording.js]: Suggestion: "returns a recording in a JSON format" => "returns metadata for a recording in JSON format"

File: api/V1/Recording.js [hunk 3]
```diff
@@ -216,4 +217,80 @@ module.exports = (app, baseUrl) => {
...
+    * Web Token (JWT) which can be used to retrieve a specific audio recording.
```
Comment (mjs) [inline api/V1/Recording.js]: Not just audio right? How about "... can be used to retrieve the recorded content."

File: api/V1/Recording.js [hunk 4]
```diff
@@ -216,4 +217,80 @@ module.exports = (app, baseUrl) => {
...
+        // Query database.
+        var recording = await models.Recording.findOne(query);
+
+        // Check the the processing is finished.
+        if (recording.get('processingState') !== 'FINISHED')
+          return responseUtil.send(response, {
+            status_code: 204,
+            success: false,
+            messages: ["File has not been fully processed yet."],
+          });
```
Comment (mjs) [inline api/V1/Recording.js]: Is there a way to retrieve the original upload vs the final processed version via the API? We'll need this for the thermal videos where people may want to look at the processed video, but we'll want the raw CPTV file for training the machine learning model.

Comment (CameronRP) [reply api/V1/Recording.js]: I was going to add this at a later date but can add it now if you want? We will need to make sure that you can only get raw data from approved data types also so will need to add the audio recordings types if we add it now.

File: api/V1/Recording.js (hunk 3 above)
Comment (CameronRP) [reply api/V1/Recording.js]: Yes! I'll change that..

**Comment by mjs** [comment]
Looks great.

**Pull request merged**